app/models/king.rb

```diff
@@ -1,19 +1,56 @@
 # rubocop:disable Metrics/AbcSize
 # rubocop:disable Style/SymbolProc
 class King < Piece
+  @rook_id = nil # used by piece to get castled rook ID
```

Comment:
Instance variables are automatically nil and don't need to be declared, so you don't need this here (kinda weird, but that's how it is)